server/src/main/java/org/apache/iotdb/db/mpp/operator/meta/TimeSeriesMetaScanOperator.java

```diff
@@ -0,0 +1,131 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.mpp.operator.meta;
+
+import org.apache.iotdb.commons.consensus.ConsensusGroupId;
+import org.apache.iotdb.db.exception.metadata.MetadataException;
+import org.apache.iotdb.db.metadata.path.PartialPath;
+import org.apache.iotdb.db.mpp.operator.OperatorContext;
+import org.apache.iotdb.db.qp.physical.sys.ShowTimeSeriesPlan;
+import org.apache.iotdb.db.query.dataset.ShowTimeSeriesResult;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.read.common.block.TsBlock;
+import org.apache.iotdb.tsfile.read.common.block.TsBlockBuilder;
+import org.apache.iotdb.tsfile.utils.Binary;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class TimeSeriesMetaScanOperator extends MetaScanOperator {
+  private String key;
+  private String value;
+  private boolean isContains;
+
+  // if is true, the result will be sorted according to the inserting frequency of the timeseries
+  private boolean orderByHeat;
+
+  private static final TSDataType[] resourceTypes = {
+    TSDataType.TEXT,
+    TSDataType.TEXT,
+    TSDataType.TEXT,
+    TSDataType.TEXT,
+    TSDataType.TEXT,
+    TSDataType.TEXT,
+    TSDataType.TEXT,
+    TSDataType.TEXT
+  };
+
+  public TimeSeriesMetaScanOperator(
+      OperatorContext operatorContext,
+      ConsensusGroupId schemaRegionId,
+      int limit,
+      int offset,
+      PartialPath partialPath,
+      String key,
+      String value,
+      boolean isContains,
+      boolean orderByHeat,
+      boolean isPrefixPath,
+      List<String> columns) {
+    super(operatorContext, schemaRegionId, limit, offset, partialPath, isPrefixPath, columns);
+    this.isContains = isContains;
+    this.key = key;
+    this.value = value;
+    this.orderByHeat = orderByHeat;
+  }
+
+  public String getKey() {
+    return key;
+  }
+
+  public String getValue() {
+    return value;
+  }
+
+  public boolean isContains() {
+    return isContains;
+  }
+
+  public boolean isOrderByHeat() {
+    return orderByHeat;
+  }
+
+  @Override
+  protected TsBlock createTsBlock() throws MetadataException {
+    TsBlockBuilder builder = new TsBlockBuilder(Arrays.asList(resourceTypes));
+    schemaRegion
+        .showTimeseries(convertToPhysicalPlan(), operatorContext.getInstanceContext())
```

Comment:
1. Add a TODO in `SchemaRegion` we may totally change `showTimeseries`'s parameter from `PhysicalPlan` to something else like operator.
2. I think it's better to change the previous implementation. We should implement this method in an iterator way instead of loading all the schema at once. You can think more about how to do that.

Reply:
1. added
2. I  will do it in another pr